db/database.go

```diff
@@ -94,6 +94,10 @@ type Backend interface {
     // Can return nil if the submission does not exist.
     GetSubmissionResult(assignment *model.Assignment, email string, shortSubmissionID string) (*model.GradingInfo, error);
 
+    // Get the result history for a specific user.
+    // An assignment must be specified.
```

Comment:
No need to specify the part about the assignment.
All these methods may take pointers (for efficiency and consistency), but none (I think) allow courses or assignments to be nil/optional.
Putting that comment would actually make the rest of the documentation more confusing (since it would put the nil-ability of the rest of the arguments in doubt).

Reply:
I resolved through a commit, but, if it is ok with you, I might leave it unresolved until we come up with a naming scheme for the API endpoint.  I think the documentation should maybe reflect what we come up with?